src/main/scala/is/hail/expr/Relational.scala

```diff
@@ -2384,3 +2388,366 @@ case class MatrixFilterEntries(child: MatrixIR, pred: IR) extends MatrixIR {
     mv.copy(rvd = newRVD)
   }
 }
+
+// follows key_by non-empty key
+case class TableAggregateByKey(child: TableIR, expr: IR) extends TableIR {
+  require(child.typ.keyOrEmpty.nonEmpty)
+
+  def children: IndexedSeq[BaseIR] = Array(child, expr)
+  
+  def copy(newChildren: IndexedSeq[BaseIR]): TableAggregateByKey = newChildren match {
+    case Seq(child: TableIR, _: IR) =>
+      TableAggregateByKey(child, expr)
```

Comment:
the copy should use the new expr IR. This is also wrong for MatrixTableAggregateRowsByKey. I'll fix it in my branch. Also add the assert `assert(newChildren.length == 2)`